### PR TITLE
Updates to README.md

### PR DESCRIPTION
**Pull request by craigmac**
Professional opinion (fulltime technical writer):
* Markdown linter
* Once line per sentence rule (md doesn't care it formats exactly the same but easier to read and review)
* Avoid passive voice and use spellchecker
* Be more specific, enumerate where possible
* Consistency among examples: start with pyright example then continue using pyright and not switch to rls.
* Consistency of naming
* etc, etc. ..

I'd consider this a first pass, there's lots more to be done in terms of structure here, I think, but it's getting there.

### CODE REVIEW

**Comment by mjlbach** [comment]
Just so you know, lspconfig.txt is auto-generated so your changes will be overwritten on merge :)

File: README.md [hunk 1]
```diff
@@ -116,61 +154,77 @@ for _, lsp in ipairs(servers) do
...
 
 1. Read [CONTRIBUTING.md](CONTRIBUTING.md).
-2. Choose a language from [the coc.nvim wiki](https://github.com/neoclide/coc.nvim/wiki/Language-servers) or
-  [emacs-lsp](https://github.com/emacs-lsp/lsp-mode#supported-languages).
-3. Create a new file at `lua/lspconfig/SERVER_NAME.lua`.
-   - Copy an [existing config](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/)
-     to get started. Most configs are simple. For an extensive example see
-     [texlab.lua](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/texlab.lua).
-4. Ask questions on our [Discourse](https://neovim.discourse.group/c/7-category/7) or in the [Neovim Gitter](https://gitter.im/neovim/neovim).
+1. Choose a language from [the coc.nvim wiki](https://github.com/neoclide/coc.nvim/wiki/Language-servers) or [emacs-lsp](https://github.com/emacs-lsp/lsp-mode#supported-languages).
```
Comment (mjlbach) [inline README.md:225]: Is there a reason you changed this to repeat 1.?

File: README.md [hunk 2]
```diff
@@ -116,61 +154,77 @@ for _, lsp in ipairs(servers) do
...
+   nvim-lspconfig does not install language servers for you.
+   You should be able to run the `cmd` defined in each server's lua module from the command line and see that the language server starts.
+   If the `cmd` is an executable name, ensure its location is listed in your system's `PATH` variable.
+1. Not triggering root detection.
+   The language server will only start if it is opened in a directory, or child directory, containing a file which signals the _root_ of the project.
+   Most of the time, this is a `.git` folder, but a server may define the root differently: see [CONFIG.md](CONFIG.md) or the source for the list of root directories.
 
-:LspInfo provides a handy overview of your active and configured language servers.  Note, that it will not report any configuration changes applied in `on_new_config`.
+`:LspInfo` provides a handy overview of your active and configured language servers.
+However, `:LspInfo` can be inaccurate if any configuration changes were applied in the `on_new_config()` hook.
```
Comment (mjlbach) [inline README.md:173]: Inaccurate is a strong word, I like the old wording better-ish

`:LspInfo` will not report any configuration changes applied in the `on_new_config` hook.

File: README.md [hunk 3]
```diff
@@ -1,81 +1,119 @@
...
+
+For a full list of servers, see [CONFIG.md](CONFIG.md).
+This document contains installation instructions and optional customizations for each language server.
+For some servers that are not on your system path, e.g., `jdtls`, `elixirls`, you will be required to manually add `cmd` as an entry in the table passed to setup.
 
 ## Keybindings and completion
 
-nvim-lspconfig does not map keybindings or enable completion by default. Manual, triggered completion can be provided by neovim's built-in omnifunc. For autocompletion, a general purpose [autocompletion plugin](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) is required. The following example configuration provides suggested keymaps for the most commonly used language server functions, and manually triggered completion with omnifunc (\<c-x\>\<c-o\>). 
+nvim-lspconfig does not map keybindings or enable completion by default.
+Manually-triggered completion can be provided by Neovim's built-in [https://neovim.io/doc/user/options.html#'omnifunc'](omnifunc).
```
Comment (mjlbach) [inline README.md:105]: don't think links usually follow the word?

File: README.md [hunk 4]
```diff
@@ -1,81 +1,119 @@
...
+For autocompletion, a general purpose [autocompletion plugin](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) is required.
 
-```vim
+Copy the following example configuration into your `init.vim` file.
```
Comment (mjlbach) [inline README.md:108]: Seems a little imperative, the example config isn't required by any means, it's just a suggestion

File: README.md [hunk 5]
```diff
@@ -1,81 +1,119 @@
...
+The configuration provides keymaps for the most commonly used language server functions.
```
Comment (mjlbach) [inline README.md:109]: I would still say following

File: README.md [hunk 6]
```diff
@@ -1,81 +1,119 @@
...
+Enabling a language server means:
+
+1. Installing the language server binary: `npm install global pyright`;
+1. Ensuring the language server can be found on your `PATH`: typing `pyright --version` succeeds;
+1. Adding the language server configuration to your `init.vim` file:
 
 ```vim
 lua << EOF
-require'lspconfig'.rust_analyzer.setup{}
+require'lspconfig'.pyright.setup{}
```
Comment (mjlbach) [inline README.md:94]: I would like to use a different language server here, as we use pyright in the example conifg.

File: README.md [hunk 7]
```diff
@@ -1,81 +1,119 @@
...
+- `requirements.txt`.
 
-Language servers require each project to have a `root` in order to provide completion and search across symbols that may not be defined in your current file, and to avoid having to index your entire filesystem on each startup.
+Language servers require each project to have a `root` in order to provide completion and search across symbols that may not be defined in your current file; it also avoids having to index your entire filesystem on each startup.
 
 ## Enabling additional language servers
 
-Enabling most language servers is as easy as installing the language server, ensuring it is on your PATH, and adding the following to your config:
+Most language servers can be installed in less than a minute.
+Enabling a language server means:
```
Comment (mjlbach) [inline README.md:86]: don't know how I feel about  `Enabling a language server means:`

File: README.md [hunk 8]
```diff
@@ -1,81 +1,119 @@
...
+
+1. Installing the language server binary: `npm install global pyright`;
+1. Ensuring the language server can be found on your `PATH`: typing `pyright --version` succeeds;
```
Comment (mjlbach) [inline README.md:89]: This whole thing is kinda a dupe of the quickstart

File: README.md [hunk 9]
```diff
@@ -1,81 +1,119 @@
...
 ## Automatically launching language servers
 
-In order to automatically launch a language server, lspconfig searches up the directory tree from your current buffer to find a file matching the `root_dir` pattern defined in each server's configuration.  For [pyright](CONFIG.md#pyright), this is any directory containing ".git", "setup.py",  "setup.cfg", "pyproject.toml", or "requirements.txt").
+In order to automatically launch a language server, lspconfig searches up the directory tree from your current buffer to find a file matching the `root_dir` pattern defined in each server's configuration.
+For [pyright](CONFIG.md#pyright), this is any directory containing one of:
+
+- `.git`,
+- `setup.py`,
+- `setup.cfg`,
+- `pyproject.toml`, or
```
Comment (mjlbach) [inline README.md:78]: If we're doing a list should we get rid of the commas/or?

File: README.md [hunk 10]
```diff
@@ -1,81 +1,119 @@
...
+1. Install a language server, e.g., [pyright](CONFIG.md#pyright)
+
 ```bash
 npm i -g pyright
 ```
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [CONFIG.md](CONFIG.md)
+
+1. Add the language server setup to your `init.vim`.
+   The server name must match those found in the table of contents in [CONFIG.md](CONFIG.md).
+   _NOTE_: this snippet is for `init.vim` only, this will not work in a `init.lua` file.
```
Comment (mjlbach) [inline README.md:46]: It will, if you remove the heredoc, we should make that clear without getting too wordy

File: README.md [hunk 11]
```diff
@@ -1,81 +1,119 @@
...
 - completion
 - rename
-- format 
+- format
 - refactor
 
-Neovim provides an interface for all of these features, and the language server client is designed to be highly extensible to allow plugins to integrate language server features which are not yet present in Neovim core such as [**auto**-completion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) (as opposed to manual completion with omnifunc) and [snippet integration](https://github.com/neovim/nvim-lspconfig/wiki/Snippets).
+Neovim itself provides an interface for all of these features.
+The LSP client is designed to be highly extensible; it allows plugins to integrate language server features which are not yet present in Neovim core.
+Examples of language server features not yet in Neovim core:
```
Comment (mjlbach) [inline README.md:22]: autocompletion and snippets (ok, maybe arguably for snippets) aren't language server features, but features of how a client can interact with data returned by the language server.

File: README.md [hunk 12]
```diff
@@ -1,81 +1,119 @@
...
 - find-references
 - hover
 - completion
 - rename
-- format 
+- format
 - refactor
 
-Neovim provides an interface for all of these features, and the language server client is designed to be highly extensible to allow plugins to integrate language server features which are not yet present in Neovim core such as [**auto**-completion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) (as opposed to manual completion with omnifunc) and [snippet integration](https://github.com/neovim/nvim-lspconfig/wiki/Snippets).
+Neovim itself provides an interface for all of these features.
```
Comment (mjlbach) [inline README.md:20]: itself -> core

**Comment by mjlbach** [comment]
Overall looks good, left some remarks :)

File: README.md (hunk 1 above)
Comment (craigmac) [reply README.md:225]: Just standard markdownlint fix: they will auto-update so there's no need to explicitly say step numbers.

File: README.md (hunk 2 above)
Comment (craigmac) [reply README.md:173]: Sounds good to me.

File: README.md (hunk 3 above)
Comment (craigmac) [reply README.md:105]: oops! yep that's the wrong order

File: README.md (hunk 1 above)
Comment (mjlbach) [reply README.md:225]: Isn't this confusing when people open markdown with a text editor like vim/neovim that doesn't transparently render the numbers?

File: README.md (hunk 4 above)
Comment (craigmac) [reply README.md:108]: I wanted to make sure to init.vim to highlight that this is for .vim files only, init.lua this would not work. Either that, or we include init.lua examples as well for each?

File: README.md (hunk 7 above)
Comment (craigmac) [reply README.md:86]:
```suggestion
To enable a language server, you need to:
```

File: README.md (hunk 8 above)
Comment (craigmac) [reply README.md:89]: Yes, I think so too, could probably be removed completely.

File: README.md (hunk 5 above)
Comment (craigmac) [inline README.md:109]:
```suggestion
The following configuration provides keymaps for the most commonly used language server functions.
```

File: README.md (hunk 1 above)
Comment (craigmac) [reply README.md:225]: Could be I guess, I wouldn't do it that way but it's your repo.

File: README.md (hunk 4 above)
Comment (mjlbach) [reply README.md:108]: init.vim/init.lua examples are identical with the exception of the lua << EOF/ EOF pairing

**Review by craigmac — COMMENTED** [review]
Small changes.

File: README.md (hunk 8 above)
Comment (mjlbach) [reply README.md:89]: I think we need to include is as people have shown that they do not generalize from the quickstart to installing another language server, I pick rust analyzer due to popularity, but tsserver is probably the most common one

File: README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (mjlbach) [reply README.md:22]: I still don't like the "not yet in" neovim core as we're unlikely to add autocompletion

**Comment by mjlbach** [comment]
I'd go ahead and commit all of the suggested changes you made, squash, and I'll take another look this weekend :) Thanks!

**Comment by mjlbach** [comment]
Hi @craigmac, if you have time could you go ahead and commit/squash everything, I'll re-review, and we can merge?

**Comment by craigmac** [comment]
Sorry, I'm not sure what you need me to do at this point? It's been awhile, and I don't have any time right now to do anything with this.

**Comment by lithammer** [comment]
> Sorry, I'm not sure what you need me to do at this point? It's been awhile, and I don't have any time right now to do anything with this.

You have a few merge conflicts to handle.
